### PR TITLE
libckteec: debug helpers for PKCS#11 IDs as strings

### PR DESCRIPTION
**Pull request by etienne-lms**
ck_debug.c implements other of such functions to trace PKCS#11
IDs from their label rather than from their numerical value.

Exported header file ck_debug.h describes the functions provided
by libckteec. The functions are label ck<something>2str(<arglist>).

First helper ckr2str() converts PKCS#11 error codes CKR_* into strings.

### CODE REVIEW

File: libckteec/src/ck_debug.c [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+	CK2STR_ENTRY(CKR_EXCEEDED_MAX_ITERATIONS),
+	CK2STR_ENTRY(CKR_FIPS_SELF_TEST_FAILED),
+	CK2STR_ENTRY(CKR_LIBRARY_LOAD_FAILED),
+	CK2STR_ENTRY(CKR_PIN_TOO_WEAK),
+	CK2STR_ENTRY(CKR_PUBLIC_KEY_INVALID),
+	CK2STR_ENTRY(CKR_FUNCTION_REJECTED),
+	CK2STR_ENTRY(CKR_VENDOR_DEFINED),
+};
+
+const char *ckr2str(CK_RV id)
```
Comment (jbech-linaro) [inline libckteec/src/ck_debug.c]: Nothing wrong with this (ckr2str_table + ckr2str), but I think this version is a bit shorter

```c
#define CK2STR_ENTRY(id) case id: return #id

const char *ckr2str(CK_RV id)
{
    switch (id) {                                                                                                                                                                                                                    
        	CK2STR_ENTRY(CKR_OK);
	        CK2STR_ENTRY(CKR_CANCEL);
                ...
                default:                                                                                                                                                                                                                             
                return "Unknown id"; 
     }
}
```

No strong opinion, just giving a suggestion for a shorter alternative.

Comment (etienne-lms) [reply libckteec/src/ck_debug.c]: Thanks for the suggestion. Looks ok to me but I wonder if it is fine to use macros for such piece of code "`case XXX: ... `".
If you are all fine with this, i'll use your proposal.

Comment (jforissier) [reply libckteec/src/ck_debug.c]: Looks reasonable to me

Comment (etienne-lms) [reply libckteec/src/ck_debug.c]: ok, thanks Jérôme.

**Comment by etienne-lms** [comment]
Comments addressed.

Checkpatch complains, consider as a false positive:
```
776374a [review] libckteec: debug helpers for PKCS#11 IDs as strings
ERROR: Macros with complex values should be enclosed in parentheses
#27: FILE: libckteec/src/ck_debug.c:14:
+#define CK2STR_ENTRY(id)	case id: return #id

WARNING: Macros with flow control statements should be avoided
#27: FILE: libckteec/src/ck_debug.c:14:
+#define CK2STR_ENTRY(id)	case id: return #id

WARNING: Possible switch case/default not preceded by break or fallthrough comment
#235: FILE: libckteec/src/ck_debug.c:116:
+	default:

total: 1 errors, 2 warnings, 0 checks, 229 lines checked
```

**Comment by etienne-lms** [comment]
Note: I added a commit to use `ckr2str()` from `ckteec_assert_expected_rv()`.

**Comment by etienne-lms** [comment]
ping

**Comment by jbech-linaro** [comment]
`Reviewed-by: Joakim Bech <joakim.bech@linaro.org>`

**Comment by etienne-lms** [comment]
fixup commit squashed, tags applied.
thanks.

**Pull request merged**